src/state/opamStateConfig.mli

```diff
@@ -27,6 +27,11 @@ type t = private {
   unlock_base: bool;
   no_env_notice: bool;
   locked: string option;
+  ignore_depext: bool;
+  depext_ignore_opam : bool;
+  depext_no_sudo : bool;
+  depext_not_root : bool;
+  depext_bypass: OpamSysPkg.Set.t * OpamSysPkg.Set.t;
```

Comment:
In the state config, we still need to have both sets, in order to have the final set at switch load (cf. https://github.com/ocaml/opam/pull/3975/commits/ba5cede46cf39674e99552fef437143d6680bbae#diff-ba849feebcfb8069706c5492d1a09521R280)

Reply:
I don't like using globals for that (they are for pervasive options that can be used throughout and would be really heavy to carry everywhere); here we are just being lazy, and using them to pass a hidden argument to `OpamSwitchState.load`.

This said, the way the code is now it's true that there is no nice way to do it :/
At least not for switches (the global config is handled differently, we can read it before applying the options).
So if we decide to make the setting global, it would be much easier.

Another possibility is that we move it to a separate command (in a sense, it would make more sense, it is a little weird to have to do an `opam install` to remove something from the bypasses; this said, we have a similar issue with `--add-to-roots`, and in practice it's ok). That might also help with the effectful nature of the option.

Reply:
Those options are now in global config